### PR TITLE
Add parentheses when necessary in transmute suggestion (fixes #1049)

### LINKED ISSUE #1049: Invalid suggestion for fix in transmute_ptr_to_ref
I have the following line, where clippy correctly warns about needless transmute from pointer to reference
```rust
let data: &mut HandlerData<T> = unsafe { mem::transmute(instance_ptr as *mut HandlerData<T>) };
```
Clippy suggests the following fix, but it needs additional brackets to successfully compile:
```rust
// suggested by clippy, but doesn't work
let data: &mut HandlerData<T> = unsafe { &mut *instance_ptr as *mut HandlerData<T> };
// should instead be
let data: &mut HandlerData<T> = unsafe { &mut *(instance_ptr as *mut HandlerData<T>) };
```

### PR DESCRIPTION
**Pull request by Manishearth**
r? @llogiq @mcarton

### CODE REVIEW

File: clippy_lints/src/transmute.rs [hunk 1]
```diff
@@ -122,7 +122,12 @@ impl LateLintPass for Transmute {
 
 
                                     let sugg = if from_pty.ty == to_rty.ty {
-                                        format!("{}{}", deref, arg)
+                                        // Put things in parentheses if they are more complex
+                                        if let ExprPath(..) = args[0].node {
```
Comment (oli-obk) [inline clippy_lints/src/transmute.rs]: This should check for more. `ExprBlock`, `ExprCall`, ... also don't need parens

Comment (Manishearth) [inline clippy_lints/src/transmute.rs]: Would be nice if we had a precendence tool that could figure this out on its own 😄

Comment (Manishearth) [inline clippy_lints/src/transmute.rs]: done

Comment (oli-obk) [inline clippy_lints/src/transmute.rs]: filed precedence tool idea under #1052

**Pull request merged**